components/fonts/platform/freetype/font.rs

```diff
@@ -106,12 +105,11 @@ fn create_face(data: Arc<Vec<u8>>, face_index: u32) -> Result<FT_Face, &'static
         let library = FreeTypeLibraryHandle::get().lock();
 
         // This is to support 32bit Android where FT_Long is defined as i32.
```

Comment:
The comment notes that the conversion is required for Android. Perhaps, we just need a cfg(target_os = "android") here?

Reply:
Good point, we have both aarch64 and armv7 builds, so we'll need to include `target_pointer_width` as well in the `cfg`.

Reply:
I've used `as FT_Long` which seems to both compile on 32-bit Android and also not cause any clippy warnings.

Reply:
I believe using `as` will just [reinterpret](https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast) the bits of the `u32` value as `i32`, which means it is possible to have an index that silently gets interpreted a negative value.  According to the FreeType documentation, negative values [have special meaning](https://freetype.org/freetype2/docs/reference/ft2-face_creation.html#ft_open_face). I haven't fully grokked the semantics of that parameter, so I'm not sure how serious of an issue this is in practice. 

Clippy doesn't warn about this because `as` is safe in some cases but not in others, so it is [allow by default](https://rust-lang.github.io/rust-clippy/master/#/as_) and specific cases need to be enabled explicitly.

Reply:
In this case the parameter is the `face_index` that we received from FontConfig. For it to be interpreted as a negative number the face index would need to be very large, correct. FontConfig should be getting this data from FreeType (converting it from i32 to u32). It seems like the only way this could happen on 32-bit platforms is if there is some kind of data corruption.

Additionally, on Android the `font_index` is always 0. See https://github.com/servo/servo/blob/93fdb8263d14346d0757c2192527bc8c7c577572/components/fonts/platform/freetype/android/font_list.rs#L33. This is the value that is ultimately passed to this function.

Reply:
Ah, I didn't realize it was hardcoded to `0` on Android. I'm fine with using `as` in this instance, but it seems like in general we should enable those clippy lints to catch such cases. Using `as` based on the implementation assumptions that could change (e.g on Android we could switch to different API for font enumeration in the future) seems like it could cause subtle issues in the long run.